### DISCUSSION TITLE
ZKsync deployment

### DISCUSSION THREAD

**Discussion by Raphmola** [Cyfrin/foundry-full-course-cu#2890] [Q&A]
after activating the plugin on remix i can't compile my simplestorage contract am stuck here need help

**Comment by EngrPips** [comment]
Hello @Raphmola, what error are you getting?

**Reply by Raphmola** [reply]
thanks for replying I have solved that problem I rewatched the video and discovered my mistake, but am having another issue now 
Please the picture below
![compiling error](https://github.com/user-attachments/assets/116ea663-cdc3-423e-a460-e764b5244e9d)

**Reply by Raphmola** [reply]
Error in plugin @matterlabs/hardhat-zksync-verify: The address provided as argument contains a contract, but its bytecode doesn't match any of your local contracts.

      Possible causes are:
        - Contract code changed after the deployment was executed. This includes code for seemingly unrelated contracts.
        - A solidity file was added, moved, deleted or renamed after the deployment was executed. This includes files for seemingly unrelated contracts.
        - Solidity compiler settings were modified after the deployment was executed (like the optimizer, target EVM, etc.).
        - The given address is wrong.
        - The selected network is wrong.
ZkSyncVerifyPluginError: The address provided as argument contains a contract, but its bytecode doesn't match any of your local contracts.

      Possible causes are:
        - Contract code changed after the deployment was executed. This includes code for seemingly unrelated contracts.
        - A solidity file was added, moved, deleted or renamed after the deployment was executed. This includes files for seemingly unrelated contracts.
        - Solidity compiler settings were modified after the deployment was executed (like the optimizer, target EVM, etc.).
        - The given address is wrong.
        - The selected network is wrong.
    at inferContractArtifacts (/opt/app/api/hardhat_env/node_modules/@matterlabs/hardhat-zksync-verify/src/plugin.ts:48:45)
    at async OverriddenTaskDefinition.getContractInfo [as _action] (/opt/app/api/hardhat_env/node_modules/@matterlabs/hardhat-zksync-verify/src/task-actions.ts:310:31)
    at async Environment._runTaskDefinition (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:359:14)
    at async Environment.run (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:192:14)
    at async OverriddenTaskDefinition.verifyContract [as _action] (/opt/app/api/hardhat_env/node_modules/@matterlabs/hardhat-zksync-verify/src/task-actions.ts:197:54)
    at async Environment._runTaskDefinition (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:359:14)
    at async Environment.run (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:192:14)
    at async OverriddenTaskDefinition.verify [as _action] (/opt/app/api/hardhat_env/node_modules/@matterlabs/hardhat-zksync-verify/src/task-actions.ts:83:5)
    at async Environment._runTaskDefinition (/opt/app/api/hardhat_env/node_modules/hardhat/src/internal/core/runtime-environment.ts:359:14)
    at async Environment.run (/opt       


(it showing the error)

**Comment by RycBrownrigg** [comment]
If you watch the video adter the ZKSync deployment, you will see that the contract needs to be in a folder named 'contracts'.  If you, make a folder/directory called contracts, if you need to, and move your contract into it, it should work.  It worked for me.

**Reply by Raphmola** [reply]
have done that already and giving the same error
![contra t](https://github.com/user-attachments/assets/8347e20a-6e84-46a3-a9fb-5b29618883f1)
![error](https://github.com/user-attachments/assets/6983b9ca-061c-44b2-abfe-1531a49f890f)

**Reply by Raphmola** [reply]
![deploy error](https://github.com/user-attachments/assets/54a23742-a368-4551-8649-dba92b8dff91)
that what am seeing, and please how will i remove remixAI space it takinh up space

**Reply by EngrPips** [reply]
There is an arrow down on the top part of the remixAI you can close it with that. Also, uncheck the `verify` box and retry the deployment process

**Reply by AlexanderCGKarlsson** [reply]
@Raphmola 

Can you try with solidity version 0.8.24?

I just did the deployment and it looks like this:

```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.24;


contract SimpleStorage {
    
    // Variables
    uint256 myfavoriteNumber;

    
    struct Person {
        uint256 favoriteNumber;
        string name;
    }

    // Arrays
    Person[] public listOfPeople;

    // Mappings
    mapping(string => uint256) public nameToFavoriteNumber; 

    function store(uint256 _favoriteNumber) public {
        myfavoriteNumber = _favoriteNumber;
    }

    function retrieve() public view returns (uint256) {
        return myfavoriteNumber;
    }


    function addPerson(string memory _name, uint256 _favoriteNumber) public {
        listOfPeople.push(Person(_favoriteNumber, _name));
        nameToFavoriteNumber[_name] = _favoriteNumber; // Automatically makes it that people can check for the NAME and get their favorite number.
    }   
}
``` 

So instead of using 0.8.26 (It looks like you are using it in the above code snippet) set it to 0.8.24.

**Reply by EngrPips** [reply]
Great heads-up
